File: 1935.可以输入的最大单词数.java
/*
 * @lc app=leetcode.cn id=1935 lang=java
 *
 * [1935] 可以输入的最大单词数
 */

// @lc code=start
class Solution {
    public int canBeTypedWords(String text, String brokenLetters) {
        // 将坏掉的字母放进一个Set里，方便查询
        Set<Character> brokenSet = new HashSet<>();
        for (char ch : brokenLetters.toCharArray()) {
            brokenSet.add(ch);
        }

        // 把整句拆成单词
        String[] words = text.split(" ");
        int count = 0;

        // 检查每个单词
        for (String word : words) {
            boolean canType = true;
            for (char ch : word.toCharArray()) {
                if (brokenSet.contains(ch)) {
                    canType = false; // 这个单词不能打
                    break;
                }
            }
            if (canType) {
                count++; // 能打，计数+1
            }
        }

        return count;
    }
}
// @lc code=end

File: 1935.可以输入的最大单词数.js
/*
 * @lc app=leetcode.cn id=1935 lang=javascript
 *
 * [1935] 可以输入的最大单词数
 */

// @lc code=start
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
var canBeTypedWords = function(text, brokenLetters) {
    const brokenSet = new Set(brokenLetters);

    const words = text.split(' ');
    let count = 0;

    for (let word of words) {
        let canType = true;

        for (let ch of word) {
            if (brokenSet.has(ch)) {
                canType = false;
                break;
            }
        }

        if (canType) {
            count++;
        }
    }

    return count;
};
// @lc code=end

